dev-clean/headers/llsc_external.hpp

```diff
@@ -67,6 +66,16 @@ inline std::string get_string(PtrVal ptr, SS state) {
   return name;
 }
 
+inline immer::flex_vector<std::pair<SS, PtrVal>> print_string(SS state, immer::flex_vector<PtrVal> args) {
```

Comment:
what's the usage of `print_string`?

Reply:
In the source file, we can use `print_string("some output")` so that the instrumented executable would print "some output" when being run. Maybe it is not the best way to do it, but I didn't find an existing way to print literal things to the screen (apart from `sym_print`, which prints a variable).

Reply:
so it seems mainly for debugging?

Reply:
Yes.

Reply:
I see, to me it seems directly adding some print statements in the generated C++ files is more straightforward to debug (you just need to run make, instead of run Scala first then make).

Reply:
Or, you can slightly change the metaprogram to generate those prints. For example, https://github.com/Kraks/sai/blob/e6962e0c7b0c0d18b881436b6109d9f699fa07f1/dev-clean/src/main/scala/sai/llsc/engines/Engine.scala#L575 will generate a `std::court << ... ` for each block-function.

Reply:
David's approach is better for printing at a specific location in a larger program. It might be hard to instrument the print statement to the proper location in the generated code. If you add print statement in metaprorgams, the problem is that it might generate c++ prints for all occurrence of the block function/instruction.